### PR TITLE
LPS-135150 Take into account the history router base path for the canonical URLs

### PR DESCRIPTION
**Pull request by LuismiBarcos**
Related ticket for the bug -> [LPS-135150](https://issues.liferay.com/browse/LPS-135150)

The ticket describes a bug with the generation of canonical URLs when configuring a history router for the Questions app.

**Steps to reproduce:**

- Add a widget page named Questions Page > place a Questions widget on it
- Set the history router as /questions-page
- Add the policy for guest access question details
- Post a question named Question
- logout
- Navigate to the details of the posted question 

**Expected Result:**
The Canonical URL is
`<link href="http://localhost:8080/web/guest/questions-page/questions/category/question" rel="canonical" data-react-helmet="true"> `

**Actual Result:**
The Canonical URL display as following, it has no page name.
`<link href="http://localhost:8080/web/guest/questions/category/question" rel="canonical" data-react-helmet="true"> `

With the fix, it into account the history router base path specified to get the full path.

### CODE REVIEW

**Comment by john-co** [comment]
Reviewed. Test failures are unrelated. :heavy_check_mark: 

Flaky issues: 
Upgrade tests: https://issues.liferay.com/browse/LRQA-69822
lpkg test: https://issues.liferay.com/browse/LRQA-66519

**Comment by john-co** [comment]
ci:test:relevant

**Comment by nhpatt** [comment]
> Add the policy for guest access question details

This should not be needed any more

**Comment by nhpatt** [comment]
This is getting hard to follow, have you tested it intensively? I do not want to break any existing logic.

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by LuismiBarcos** [comment]
ci:forward

**Comment by john-co** [comment]
ci:forward
